tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -482,12 +498,54 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
 }
 
 // getWorkerBackendPoolID gets the backend pool id of the worker node loadbalancer
-func (az *AzureProvider) getWorkerBackendPoolID() *string {
-	// We assume that BackendPoolID for worker vm follows the following format
-	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
-	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
-		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
-	return &wbpi
+func (az *AzureProvider) getWorkerBackendPoolID(ctx context.Context) (*string, error) {
+	filter := "resourceType eq 'Microsoft.Compute/virtualMachines' and substringof('worker', name)"
+	num := int32(1)
+	vmResource, err := az.getResource(ctx, filter, num)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get worker node: %s", err)
+	}
+	vmName := extractResourceName(*vmResource[0].ID)
```

Comment:
Why the zeroth entry?

Reply:
What if you retrieve more than one VM?

 What if you retrieve zero VMs? Is that possible? This will cause a panic if it is